etn-ccis/blui-react-workflows
# Fix/update i18next version

Fixes 
https://github.com/etn-ccis/blui-react-workflows/issues/233 
add line `i18nAppInstance.addResourceBundle('en', 'bluiCommon', { ACTIONS:{ CREATE_ACCOUNT: 'Register now!' }
}, true, true); ` in app.tsx of example file and check it should update text



#### Changes proposed in this Pull Request:

The app should run properly with all translations if we are not passing the i18n app instance to the i18n prop to auth context or registration context
- Custom screen translations should work in registration or auth workflow after passing the app i18n instance to context provider
`AuthContextProvider
                        actions={ProjectAuthUIActions(app)}
                        language={app.language}
                        navigate={navigate}
                        routeConfig={routes}
                        i18n={i18nAppInstance}
                        rememberMeDetails={{ email: rememberMe ? email : '', rememberMe: rememberMe }}
                    >
                        <Outlet />
                    </AuthContextProvider>`
- **Update typescript version to 5.2.2**
- **Update the Eslint version to the latest for fixing the below error**

<img width="1029" alt="Screenshot 2023-09-27 at 10 39 21 AM" src="https://github.com/etn-ccis/blui-react-workflows/assets/133877691/01b09d92-ab25-41e2-b021-ca72b19a5ced">



#### Screenshots / Screen Recording (if applicable)

-



#### To Test:

- Test all screens with translations
- Check the login page you should see the `Register Now!` text
- **You can use <CustomScreen/> in the registration workflow to see translations**



#### Any specific feedback you are looking for?

- **Typescript version is updated in this PR so please review all file changes to see what fixes I have added to remove type-checking errors.**
